lib/file_transfer_agent/remote_storage_util.js

```diff
@@ -41,15 +43,17 @@ function RemoteStorageUtil(connectionConfig) {
   * @returns {Object}
   */
   this.getForStorageType = function (type) {
+    if (client) {
+      return client;
+    }
     if (type === 'S3') {
-      return new SnowflakeS3Util(connectionConfig);
+      client = new SnowflakeS3Util(connectionConfig);
```

Comment:
why is this change needed? cannot we return as previously?

Reply:
If we return the client as before, a new instance will be created each time. In remote_storage_util, every method calls getForStorageType, which results in each function having a different Util instance. I believe this is not the behavior we want. WDYT @sfc-gh-dprzybysz?